### PR TITLE
ROMs IO to DataObjects

### LINKED ISSUE #879: Printing from ROMs
--------
Issue Description
--------
Some ROMs have additional metadata that can be useful to the user and should be printable.

Previously there was some implementation to IO straight to file; that should be transitioned to a general API for writing to DataObjects.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #879, depends on #877 (THAT should be merged first). 

##### What are the significant changes in functionality due to this change request?
Puts forward tools for ROMs to be able to print metadata, both pointwise and general, to DataObjects.  If a ROM doesn't have any information to print, it will print a message to file, and the resulting XML will indicate there's no data to print (due to implementation in the SupervisedLearning base class).

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by alfoa** [comment]
@PaulTalbot-INL I am going to review this once the other one is merged. Once we merge that, can you "rebase" this?

File: framework/DataObjects/DataSet.py [hunk 1]
```diff
@@ -125,11 +124,11 @@ def addExpectedMeta(self,keys):
     self._metavars.extend(keys)
     self._orderedVars.extend(keys)
 
-  def addMeta(self,tag,xmlDict):
+  def addMeta(self, tag, xmlDict = None, node = None):
```
Comment (joshua-cogliati-inl) [inline framework/DataObjects/DataSet.py]: For what it is worth, in PEP 8 this should be: ```def addMeta(self, tag, xmlDict=None, node=None):```

Comment (PaulTalbot-INL) [reply framework/DataObjects/DataSet.py]: Oh, thanks! I missed the exception for function keywords.

**Comment by PaulTalbot-INL** [comment]
Disabled the test
```
framework\PostProcessors\TemporalDataMiningPostProcessor\DimensionalityReduction\MultiDimensionalScaling
```
and opened issue #905.

**Comment by alfoa** [comment]
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. ***OK***
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). ***OK***
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. ***OK***
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. ***THEY PASS***
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. ***N/A***
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. ***N/A***
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. ***#879***
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?***N/A***

**Comment by alfoa** [comment]
Checklist passed...

**Pull request merged**